### PR TITLE
3 br creates user model

### LINKED ISSUE #3: Create User model
Validation test for email, association tests will be made as the belongs_to models are added.

User attributes:

first_name
last_name
email: uniqueness true
password_digest
admin(boolean) default: false

### PR DESCRIPTION
**Pull request by riverswb**
Closes #3

@njgheorghita @Sh1pley @ski-climb @tmikeschu

Adds users table, user model, factory, and test.

I set it up so that a user can be created with a first & last name, email (must be unique) and secure password,  but none of them are required because we will also be doing OAuth, phone number is required and must be unique.

I don't know what is wrong with Travis CI, my tests pass?

### CODE REVIEW

**Comment by njgheorghita** [comment]
looks good to me, merge on

**Comment by tmikeschu** [comment]
@riverswb have you looked into loading the test database environment?

```
bundler: failed to load command: rspec (/home/travis/build/tmikeschu/air_bnb_clone/vendor/bundle/ruby/2.3.0/bin/rspec)
ActiveRecord::NoDatabaseError: FATAL:  database "air_bnb_clone_test" does not exist
```

File: spec/factories.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :user do
+    first_name   Faker::Name.first_name
+    last_name    Faker::Name.last_name
+    email        Faker::Internet.unique.email
+    password     Faker::Internet.password
+    phone_number Faker::Number.unique.number(10)
```
Comment (tmikeschu) [inline spec/factories.rb:7]: Nice Fakers!

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context "validations" do
+    it "email must be unique" do
+      user = create(:user)
```
Comment (tmikeschu) [inline spec/models/user_spec.rb:6]: @riverswb we can clean this up with shoulda matchers if you want:
https://github.com/thoughtbot/shoulda-matchers

E.g.,
```
it { should validate_uniqueness_of :email }
```

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 VCR.configure do |config|
   config.cassette_library_dir = Rails.root.join("spec","fixtures","vcr_cassettes")
-  congig.hook_into :webmock
```
Comment (tmikeschu) [inline spec/rails_helper.rb:15]: Congig definitely doesn't work 😆

**Review by tmikeschu — COMMENTED** [review]
Looks good! See comment about shoulda matchers.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      invalid_user = User.create(email: user.email)
+
+      expect(user).to be_valid
+      expect(invalid_user).to be_invalid
+    end
+
+    it "phone number must be present" do
+      invalid_user = User.create(phone_number: nil)
```
Comment (tmikeschu) [inline spec/models/user_spec.rb:14]: You can do `validate_presence_of` too

**Pull request merged**